src/components/CommunityHolderGrid/CommunityHolderGridItem.tsx

```diff
@@ -0,0 +1,178 @@
+import { CouldNotRenderNftError } from 'errors/CouldNotRenderNftError';
+import colors from 'components/core/colors';
+
+type Props = {
+  holderRef: CommunityHolderGridItemFragment$key;
+};
+
+export default function CommunityHolderGridItem({ holderRef }: Props) {
+  const token = useFragment(
+    graphql`
+      fragment CommunityHolderGridItemFragment on Token {
+        dbid
+        name
+        tokenId
+        contract {
+          contractAddress {
+            address
+          }
+        }
+        collectorsNote
+        media {
+          __typename
+          ... on ImageMedia {
+            previewURLs @required(action: NONE) {
+              large @required(action: NONE)
+            }
+          }
+          ... on VideoMedia {
+            previewURLs @required(action: NONE) {
+              large @required(action: NONE)
+            }
+          }
+          ... on UnknownMedia {
+            previewURLs @required(action: NONE) {
+              large @required(action: NONE)
+            }
+          }
+        }
+        owner @required(action: THROW) {
+          username @required(action: THROW)
+          universal
+          ...walletTruncateUniversalUsernameFragment
+        }
+        ...getVideoOrImageUrlForNftPreviewFragment
+        ...TokenDetailViewFragment
+        ...NftPreviewAssetFragment
+      }
+    `,
+    holderRef
+  );
+
+  const [isFailedToLoad, setIsFailedToLoad] = useState(false);
+  const handleFailedToLoad = useCallback(() => {
+    setIsFailedToLoad(true);
+  }, []);
+
+  const { showModal } = useModalActions();
+
+  const { tokenId, contract, owner } = token;
+
+  const usernameWithFallback = owner ? graphqlTruncateUniversalUsername(owner) : null;
+
+  const openseaProfileLink = `https://opensea.io/${owner?.username}`;
+
+  const reportError = useReportError();
+  const previewUrlSet = getVideoOrImageUrlForNftPreview(token, reportError);
+
+  if (!previewUrlSet?.urls.large) {
+    throw new CouldNotRenderNftError('CommunityHolderGridItem', 'could not find large image url');
+  }
+
+  const openSeaExternalUrl = useMemo(() => {
+    if (contract?.contractAddress?.address && tokenId) {
+      return getOpenseaExternalUrl(contract.contractAddress.address, tokenId);
+    }
+
+    return '';
+  }, [contract?.contractAddress?.address, tokenId]);
+
+  const handleClick = useCallback(() => {
+    if (owner?.universal) {
+      window.open(openSeaExternalUrl, '_blank');
+      return;
+    }
+
+    showModal({
+      content: (
+        <StyledNftDetailViewPopover justify="center" align="center">
+          <TokenDetailView authenticatedUserOwnsAsset={false} queryRef={token} />
+        </StyledNftDetailViewPopover>
+      ),
+      isFullPage: true,
+    });
+  }, [openSeaExternalUrl, owner?.universal, showModal, token]);
+
+  const previewAsset = useMemo(() => {
+    if (isFailedToLoad || !previewUrlSet?.urls.large) {
```

Comment:
can we use `NftFailureBoundary` like the other places in the app or is this fundamentally different?

Reply:
@tbezman I think we can! Not really sure I why I didn't end up implement it 🤦🏻‍♂️. 

I'll add this `NftFailureBoundary` after the AG launch to avoid any regression.